pkg/queue/cluster_queue_best_effort_fifo_test.go

```diff
@@ -104,34 +114,50 @@ func TestClusterQueueBestEffortFIFO(t *testing.T) {
 
 			for _, w := range test.workloadsToAdd {
 				cq.PushOrUpdate(w)
+				qImpl.AddOrUpdate(w)
 			}
 
 			for _, w := range test.inadmissibleWorkloadsToRequeue {
 				cq.RequeueIfNotPresent(w, false)
+				qImpl.AddIfNotPresent(w)
 			}
 			for _, w := range test.admissibleWorkloadsToRequeue {
 				cq.RequeueIfNotPresent(w, true)
+				qImpl.AddIfNotPresent(w)
 			}
 
 			for _, w := range test.workloadsToUpdate {
 				cq.PushOrUpdate(w)
+				qImpl.AddOrUpdate(w)
 			}
 
 			for _, w := range test.workloadsToDelete {
 				cq.Delete(w)
+				delete(qImpl.items, workload.Key(w))
 			}
 
 			if test.queueInadmissibleWorkloads {
 				cq.QueueInadmissibleWorkloads()
 			}
 
+			if test.deleteFromQueue {
+				cq.DeleteFromQueue(qImpl)
+			}
+
 			gotWorkloads, _ := cq.Dump()
 			if diff := cmp.Diff(test.wantActiveWorkloads, gotWorkloads); diff != "" {
 				t.Errorf("Unexpected items in cluster foo (-want,+got):\n%s", diff)
 			}
-			if got := cq.Pending(); got != test.wantPending {
+
+			got := cq.Pending()
+			if got != test.wantPending {
 				t.Errorf("Got %d pending workloads, want %d", got, test.wantPending)
 			}
+
+			diff := got - int32(len(gotWorkloads))
```

Comment:
I'm thinking if we should add an `PendingInadmissibleWorkloads` method for `ClusterQueue`, at least for now, this can be achieved with such a trick.

Reply:
I don't think there is a need to do this check. We already counted the total and the active queues.